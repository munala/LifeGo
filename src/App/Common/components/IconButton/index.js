import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@material-ui/core/Icon';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';

import './styles.css';

const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
  },
});

const IconButton = ({
  name,
  classes,
  style,
  onClick,
  disabled,
}) => (
  <div className="container" >
    <Button
      className={classes.button}
      color="primary"
      onClick={onClick}
      disabled={disabled}
      style={style}
    >
      <Icon>{name}</Icon>
    </Button>
  </div>
);

IconButton.propTypes = {
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  style: PropTypes.shape({}),
  classes: PropTypes.shape({}).isRequired,
};

IconButton.defaultProps = {
  disabled: false,
  style: undefined,
};

export default withStyles(styles)(IconButton);
