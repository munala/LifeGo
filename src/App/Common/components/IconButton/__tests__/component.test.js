import React from 'react';
import { shallow } from 'enzyme';
import IconButton from '../';

describe('IconButton tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<IconButton
      name="home"
      onClick={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
