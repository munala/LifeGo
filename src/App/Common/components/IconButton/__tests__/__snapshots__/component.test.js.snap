// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton tests renders without crashing 1`] = `
<IconButton
  classes={
    Object {
      "button": "IconButton-button-1",
    }
  }
  disabled={false}
  name="home"
  onClick={[Function]}
/>
`;
